src/plugins/intel_cpu/src/nodes/kernels/jit_eltwise_common.cpp

```diff
@@ -53,9 +52,12 @@ ov::element::Type eltwise_precision_helper::get_precision(const size_t inputs_nu
         });
     };
 
-    assert(std::all_of(supported_precision_intersection.begin(),
-                       supported_precision_intersection.end(),
-                       has_same_precision));
+    OPENVINO_DEBUG_ASSERT(
```

Comment:
This probably could be just a compile time check, since we are getting precisions which are just some constant values.
Anyway, there is no reason to print size.

Reply:
The next assert (inside the for loop) should also be a debug one)

Reply:
> This probably could be just a compile time check, since we are getting precisions which are just some constant values.

Its calculation rely on runtime data, so unfortunately we cannot check it purely at compile time

> The next assert (inside the for loop) should also be a debug one)

Yes, done